src/state_point.F90

```diff
@@ -214,18 +218,59 @@ subroutine write_state_point()
              group="tallies/tally" // to_str(i), length=t % n_nuclide_bins)
         deallocate(temp_array)
 
-        ! Write number of score bins, score bins, and moment order
+        ! Write number of score bins, score bins, user score bins
         call sp % write_data(t % n_score_bins, "n_score_bins", &
              group="tallies/tally" // to_str(i))
         call sp % write_data(t % score_bins, "score_bins", &
              group="tallies/tally" // to_str(i), length=t % n_score_bins)
-        call sp % write_data(t % moment_order, "moment_order", &
-             group="tallies/tally" // to_str(i), length=t % n_score_bins)
-
-        ! Write number of user score bins
         call sp % write_data(t % n_user_score_bins, "n_user_score_bins", &
              group="tallies/tally" // to_str(i))
 
+      ! Write explicit moment order strings for each score bin
+#ifdef HDF5
+        call sp % open_group("tallies/tally" // trim(to_str(i)) // "/moments")
+        call sp % close_group()
+#endif
+
+        k = 1
+        MOMENT_LOOP: do j = 1, t % n_user_score_bins
+          select case(t % score_bins(k))
+          case (SCORE_SCATTER_N, SCORE_NU_SCATTER_N)
+            moment_name = 'P' // to_str(t % moment_order(k))
+            call sp % write_data(moment_name, "order", &
+                 group="tallies/tally" // trim(to_str(i)) // &
+                 "/moments/" // trim(to_str(k)))
```

Comment:
Could you refresh my memory on how I would change this function call to do that? I'm not really sure I see what the benefit is in making this change - it seems to me that including each one as a separate group with the order index explicitly defined is more transparent to the user.